src/helpers/web3Providers/HDWalletSafeProvider.js

```diff
@@ -219,20 +220,24 @@ class HDWalletSafeProvider extends TruffleHDWalletProvider {
       // Get Safe Module contract data
       const [txObject, ...extraArguments] = arguments
       const [txDetails, ...extraParams] = txObject.params
+
       const {
         from,
         to,
         value = 0,
         data
       } = txDetails
 
-      // const safe = this._safesByAddress[from]
-      const safeByOperator = this._safesByOperator[from]
-      const safeByAddress = this._safesByAddress[from]
-      const safe = safeByOperator || safeByAddress
-      // TODO: review how we execute transactions, whether or not
-      // we set 'from' to the operator or safe address 
-      assert(safe, 'Unknown safe with address ' + from)
+      const safe = this._safesByAddress[from]
```

Comment:
Take into account that checking the Safe exists requires **from** to always be the safe address. If we bots set **from** as the safe address that's ok, otherwise we would change it to **this._safeByOperator[from]**, where **from** is the operator's address